llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -1770,6 +1772,20 @@ bool VectorCombine::foldShuffleToIdentity(Instruction &I) {
       Worklist.push_back(GenerateInstLaneVectorFromOperand(Item, 1));
     } else if (isa<UnaryOperator>(Item[0].first)) {
       Worklist.push_back(GenerateInstLaneVectorFromOperand(Item, 0));
+    } else if (auto *II = dyn_cast<IntrinsicInst>(Item[0].first);
+               II && isTriviallyVectorizable(II->getIntrinsicID())) {
+      for (unsigned O = 0; O < II->getNumOperands() - 1; O++) {
```

Comment:
`for (unsigned O = 0, E = II->getNumOperands() - 1; O < E; O++) {`
Also, why O? I've always avoided it due to O/0/o confusion, but that's me :)

Reply:
Good point :) It was just short-hard for the operation number and I perhaps made it too short. I'll switch it over to use Op.